blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/credential/BlueOceanCredentialsProvider.java

```diff
@@ -0,0 +1,262 @@
+                                                                   @Nonnull List<DomainRequirement> domainRequirements,
+                                                                   @Nonnull CredentialsMatcher matcher) {
+        ListBoxModel result = new ListBoxModel();
+        if (itemGroup instanceof AbstractFolder) {
+            FolderPropertyImpl prop = (FolderPropertyImpl) ((AbstractFolder) itemGroup).getProperties().get(FolderPropertyImpl.class);
+            if (prop != null) {
+                result.add(DISPLAY_NAME, prop.getId());
+            }
+
+        }
+        return result;
+    }
+
+    @Override
+    @Nonnull
+    public String getDisplayName() {
+        return DISPLAY_NAME;
+    }
+
+    @Override
+    public CredentialsStore getStore(@CheckForNull ModelObject object) {
+        if(isApplicable(object)) {
+            return new FolderPropertyImpl.StoreImpl((AbstractFolder) object);
+        }
+        return null;
+    }
+
+    @Override
+    public Set<CredentialsScope> getScopes(ModelObject object) {
+        if(isApplicable(object)){
+            return ImmutableSet.of(CredentialsScope.USER);
+        }
+        return Collections.emptySet();
+    }
+
+    private boolean isApplicable(ModelObject object){
+        return object instanceof AbstractFolder &&
+                ((AbstractFolder)object).getProperties().get(FolderPropertyImpl.class) != null;
+    }
+
+    public static class FolderPropertyImpl extends AbstractFolderProperty<AbstractFolder<TopLevelItem>> {
+        private final String user;
+        private final String id;
+
+        @DataBoundConstructor
+        public FolderPropertyImpl(@Nonnull String user, @Nonnull String id) {
+            this.user = user;
+            this.id = id;
+        }
+
+        public @Nonnull String getUser() {
+            return user;
+        }
+
+        public @Nonnull String getId() {
+            return id;
+        }
+
+        @Override
+        public AbstractFolderProperty<?> reconfigure(StaplerRequest req, JSONObject form) throws FormException {
+            return this;
+        }
+
+        @Extension(optional = true)
+        public static class DescriptorImpl extends AbstractFolderPropertyDescriptor {
+        }
+
+        public static class CredentialsStoreActionImpl extends CredentialsStoreAction {
+
+            private final CredentialsStore store;
+
+            CredentialsStoreActionImpl(CredentialsStore store) {
+                this.store = store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Nonnull
+            @Override
+            public CredentialsStore getStore() {
+                return store;
+            }
+
+            /**
+             * {@inheritDoc}
+             */
+            @Override
+            public String getIconFileName() {
+                return isVisible()
+                        ? "/plugin/credentials/images/48x48/folder-store.png"
```

Comment:
The stock icon is for the "stock" FolderCredentialsProvider from the folders plugin. You need your own so that a user/admin sees that the credentials are coming from this proxy/gateway provider